cnd/src/http_api/routes.rs

```diff
@@ -148,6 +322,184 @@ pub struct BobHanEthereumHalightBitcoinState {
     pub finalized_swap: comit_ln::FinalizedSwap,
 }
 
+impl GetSwapStatus for BobHanEthereumHalightBitcoinState {
+    fn get_swap_status(&self) -> SwapStatus {
+        let ethereum_status = HtlcState::from(self.alpha_ledger_state.clone());
+        han_eth_halight_swap_status(&ethereum_status, &self.beta_ledger_state)
+    }
+}
+
+impl GetAlphaTransaction for BobHanEthereumHalightBitcoinState {
+    fn get_alpha_transaction(&self) -> Transaction {
+        Transaction::from(self.alpha_ledger_state.clone())
+    }
+}
+
+impl GetBetaTransaction for BobHanEthereumHalightBitcoinState {
+    fn get_beta_transaction(&self) -> Transaction {
+        Transaction::from(self.beta_ledger_state)
+    }
+}
+
+impl GetRole for BobHanEthereumHalightBitcoinState {
+    fn get_role(&self) -> Role {
+        Role::Bob
+    }
+}
+
+impl QuantityWei for BobHanEthereumHalightBitcoinState {
+    fn quantity_wei(&self) -> String {
+        self.finalized_swap.alpha_asset.to_wei_dec()
+    }
+}
+
+impl QuantitySatoshi for BobHanEthereumHalightBitcoinState {
+    fn quantity_satoshi(&self) -> String {
+        self.finalized_swap.beta_asset.as_sat().to_string()
+    }
+}
+
+#[derive(Debug, Serialize)]
+struct HanEthereum {
+    pub protocol: String,
+    pub quantity: String, // In Wei.
+}
+
+#[derive(Debug, Serialize)]
+struct HalightBitcoin {
+    pub protocol: String,
+    pub quantity: String, // In Satoshi.
+}
+
+#[derive(Debug, Serialize)]
+struct Transaction {
+    /// Keys are on of: "init", "deploy", "fund", "redeem", "refund".
+    /// Values are transactions.
+    transactions: HashMap<String, String>,
+    status: EscrowStatus,
+}
+
+#[derive(Debug, Clone, Copy, Serialize, PartialEq)]
+#[serde(rename_all = "SCREAMING_SNAKE_CASE")]
+enum EscrowStatus {
```

Comment:
Still not happy with these but I guess this is not the right place to discuss them.

Reply:
I also do not like this.  It seem to muddy together the two protocols (han/ethereum = halight) so I get the feeling that this is not going to be extensible. Pretty much all the code in `http_api/routes.rs` is horrible, its so entangled in the fact that its only this one protocol pair.